dvc/command/base.py

```diff
@@ -31,9 +34,14 @@ def __init__(self, args):
         from dvc.repo import Repo
         from dvc.updater import Updater
 
-        self.repo = Repo()
-        self.config = self.repo.config
         self.args = args
+        try:
+            self.repo = Repo()
+        except NotDvcRepoError:
+            if args.cmd in HEADLESS_COMMANDS:
+                return
+            raise
+        self.config = self.repo.config
```

Comment:
needed this to avoid

```
ERROR: you are not inside of a DVC repository (checked up to mount point '/')
```

when running `dvc completion` "headless" (no_scm, no_root) as required by the snap build.

Reply:
replaced with `CmdBaseNoRepo` doh